### PR TITLE
Add react-powered main payments settings screen

### LINKED ISSUE #50721: Populate react-powered main payments settings screen
Replicate the existing main payments settings screen using React, as much as possible (timeboxed to 2 days)

## Acceptance criteria

* Populate the react component created in https://github.com/woocommerce/woocommerce/issues/49927
* Pre-fill WooPayments
* Fetch payment gateways from data store
* Don't include re-order functionality
* Don't include 3-dots menu on WooPayments row

Image

### PR DESCRIPTION
**Pull request by chihsuan**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

Closes https://github.com/woocommerce/woocommerce/issues/50721.

This PR replicates the existing payment methods screen in the new React component (as much as possible).

It provides the following functionalities:

- It pre-fills the WooPayments gateway when store is eligible for it.
- Show the payment methods in a table by fetching the payment methods from the server.
- Can enable/disable payment methods.
- Show additional payment providers in the last row.

It doesn't include the following:

- The re-order functionality.
- The 3-dots menu on the WooPayments row.
- The sub menu items `WooPayments |All payment methods` when WooPayments is enabled.
- The Save button since re-order functionality is not supported

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. In a fresh site with beta tester plugin installed, go through onboarding wizard and set country to Afghanistan.
2. Go to `Tools > WCA Tester Helper` and enable the feature flag `reactify-classic-payments-settings`.
2. Go to `WooCommerce > Settings > Payments`
3. Observe the Main payments screen shows the payment methods in a table without WooPayments.
4. Change store country to United States.
5. Observe the Main payments screen shows the payment methods in a table with WooPayments.
6. Enable/disable payment methods and observe the changes.
7. Click the payment method and confirm it redirects to the corresponding settings page.
8. Click on WCPay `Install` button and confirm it redirects to the payment page after installation.
9. Go back to the payments screen and confirm the WooPayments row is updated. 
10. Observe that WooPayments banner (`Payments made simple, designed exclusively for WooCommerce stores.`) is shown

#### Before installing WCPay:

Classic:

Screenshot 2024-08-21 at 12 41 20

React:

Screenshot 2024-08-21 at 12 41 32

#### After installing WCPay:

Classic:

Screenshot 2024-08-21 at 12 41 03

React:
Screenshot 2024-08-21 at 12 40 46

### Changelog entry

-   [ ] Automatically create a changelog entry from the details below.

-   [ ] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Details</summary>

#### Significance

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>

### CODE REVIEW

File: plugins/woocommerce-admin/client/settings-payments/components/payment-method.tsx [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+		try {
+			const response = await fetch( window.woocommerce_admin.ajax_url, {
+				method: 'POST',
+				headers: {
+					'Content-Type': 'application/x-www-form-urlencoded',
+				},
+				body: new URLSearchParams( {
+					action: 'woocommerce_toggle_gateway_enabled',
+					security:
+						window.woocommerce_admin.nonces?.gateway_toggle || '',
```
Comment (ilyasfoo) [inline plugins/woocommerce-admin/client/settings-payments/components/payment-method.tsx]: I wonder if we should even try to fetch if nonce is unavailable? What do you think of adding a guard for the variable?

**Review by ilyasfoo — CHANGES_REQUESTED** [review]
Looking good so far, @chihsuan! I also found that `Recommended payment providers` is not shown when it's supposed to because I've removed it in https://github.com/woocommerce/woocommerce/commit/3ae1aca65a386bfbaacf14fbda1a8682c7b0578f. I can write a PR to re-introduce it in main settings page, let me know!

#### Issue 1: Extra payment methods

When `Stripe` plugin is installed and activated, multiple Stripe's payment methods are listed ([screenshot](https://github.com/user-attachments/assets/2af3f154-0bce-4517-ab48-fe664896f3cf)). While this did not occur with the non-react screen ([classic screenshot](https://github.com/user-attachments/assets/3b9702f2-9536-4064-8c36-83ed8792648f)).

I reproduce this with: 
1. Setting my country to Mexico
2. Install Stripe plugin (I also installed a bunch other from the "Recommended payment providers" while the feature flag is disabled). 
3. Observe the Stripe payment methods in `WooCommerce > Settings > Payments` like `Stripe Alipay`

#### Issue 2: Unable to enable payment gateway

Another issue I found that payment gateways couldn't be enabled (I believe it should navigated users to the specific payment gateway settings page).

To reproduce:

1. In a fresh site, select Mexico in onboarding
2. Go to `WooCommerce > Settings > Payments`
3. Install Stripe plugin from the "Recommended payment providers"
4. Enable the `reactify-classic-payments-settings` feature flag
2. Go to `WooCommerce > Settings > Payments`
3. Attempt to enable `Stripe – Credit Card (Stripe)`
4. Observe you're not not redirected anywhere and error in console



https://github.com/user-attachments/assets/1fd6653c-7114-47cc-8bac-b75971d2cd78

**Comment by adrianduffell** [comment]
> I also found that Recommended payment providers is not shown when it's supposed to because I've removed it in https://github.com/woocommerce/woocommerce/commit/3ae1aca65a386bfbaacf14fbda1a8682c7b0578f. I can write a PR to re-introduce it in main settings page, let me know!

@ilyasfoo @chihsuan Good catch on this. I'll check whether it's in the requirements!

**Review by moon0326 — COMMENTED** [review]
Other than what @ilyasfoo already mentioned, LGTM with a minor comment.

When I click `Install` button, it does not change to busy state unlike non-react page.


https://github.com/user-attachments/assets/29617595-075e-44f9-8da2-ae5365605166

File: plugins/woocommerce-admin/client/settings-payments/components/payment-method.tsx (hunk 1 above)
Comment (chihsuan) [reply plugins/woocommerce-admin/client/settings-payments/components/payment-method.tsx]: Good idea. Updated to check nonce in d19baf63227ad6c48da02d4dd94c7a618f853b27

**Comment by chihsuan** [comment]
> When I click `Install` button, it does not change to busy state unlike non-react page.

Good catch! Fixed in 466c246ecf2d92d3d02186762f5674175d7b0f62

**Comment by adrianduffell** [comment]
> I also found that Recommended payment providers is not shown when it's supposed to because I've removed it in https://github.com/woocommerce/woocommerce/commit/3ae1aca65a386bfbaacf14fbda1a8682c7b0578f. I can write a PR to re-introduce it in main settings page, let me know!

@ilyasfoo @chihsuan It's considered a "nice to have" (see below). How about we add a follow-up PR to re-introduce it if it's not much work?

_"Having the recommended plugins listing in place – as it is today – will greatly help us build on top of it. But I’d say the installed gateways list is more important in case you need to prioritize"._

**Comment by chihsuan** [comment]
> It's considered a "nice to have" (see below). How about we add a follow-up PR to re-introduce it if it's not much work?

Sounds good! 👍

**Comment by chihsuan** [comment]
Thanks for the detailed report! @ilyasfoo I've fixed those two issues in 97e088a0d9f0c5dc99f6dae9dcc7e4fe7d94f5d0 and 
e8a2ae71576c6d49cbfb0b1e2d3a220c081603d1

**Comment by chihsuan** [comment]
@adrianduffell 👋  I found something related to https://github.com/woocommerce/team-ghidorah/issues/349.

WooPayments uses React + JQuery script to show a confirmation modal when disabling the payment method (see screen recording below). 


They hook into the `woocommerce_admin_field_payment_gateways` action to [render a `div`](
https://github.com/Automattic/woocommerce-payments/blob/0d017f60546aa1870cac7b437b7cedad14ccbf0e/includes/admin/class-wc-payments-admin.php#L1067-L1076
), which is then [rendered as a modal using React](https://github.com/Automattic/woocommerce-payments/blob/0d017f60546aa1870cac7b437b7cedad14ccbf0e/client/payment-gateways/index.js#L12-L20).

I couldn't make it work with the react setting screen in this PR. I think WooPayments need to change something in their code to make it work:

1. WooPayments need to use a valid div tag. It's using the self-closing div tag `<div />` which is not valid and somehow breaks the React page so I have to disable their output buffer to make the page work.
2. WooPayments [listen to the JQuery `ajaxSend` event](https://github.com/Automattic/woocommerce-payments/blob/develop/client/payment-gateways/payment-gateways-confirmation.js#L90) to show the modal. I think they need to change it to listen to a native event to make it work with React. 


https://github.com/user-attachments/assets/b259bada-1cfd-45cf-94b3-b403654d5e6b

**Comment by adrianduffell** [comment]
> I found something related to https://github.com/woocommerce/team-ghidorah/issues/349

Thanks @chihsuan, good find! We can mention it in the audit but proceed without it in this PR 👍

**Comment by ilyasfoo** [comment]
> It's considered a "nice to have" (see below). How about we add a follow-up PR to re-introduce it if it's not much work?

@adrianduffell  Sounds good, it should be a quick PR, I'll attempt to do it today

File: plugins/woocommerce-admin/client/settings-payments/components/payment-method.tsx [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+			<td className="action" width="1%">
+				{ id === 'pre_install_woocommerce_payments_promotion' ? (
+					<WCPayInstallButton />
+				) : (
+					<a
+						className="button alignright"
+						aria-label={
+							enabled
+								? `Manage the "${ method_title }" payment method`
+								: `Set up the "${ method_title }" payment method`
```
Comment (ilyasfoo) [inline plugins/woocommerce-admin/client/settings-payments/components/payment-method.tsx]: These are missing translations

File: plugins/woocommerce-admin/client/settings-payments/components/payment-method.tsx [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+				>
+					<span
+						className={ `woocommerce-input-toggle ${
+							isEnabled
+								? 'woocommerce-input-toggle--enabled'
+								: 'woocommerce-input-toggle--disabled'
+						} ${
+							isLoading ? 'woocommerce-input-toggle--loading' : ''
+						}` }
+						aria-label={ `The "${ method_title }" payment method is currently ${
```
Comment (ilyasfoo) [inline plugins/woocommerce-admin/client/settings-payments/components/payment-method.tsx]: Also missing translation

**Review by ilyasfoo — APPROVED** [review]
Minor comments on translations, but tests well, pre-approving! 🚢

File: plugins/woocommerce-admin/client/settings-payments/components/payment-method.tsx (hunk 2 above)
Comment (chihsuan) [reply plugins/woocommerce-admin/client/settings-payments/components/payment-method.tsx]: Good catch! Fixed in 62a3b2209236f8966678f6da362834552213143a

File: plugins/woocommerce-admin/client/settings-payments/components/payment-method.tsx (hunk 3 above)
Comment (chihsuan) [reply plugins/woocommerce-admin/client/settings-payments/components/payment-method.tsx]: Fixed in 62a3b2209236f8966678f6da362834552213143a

**Review by moon0326 — APPROVED** [review]
LGTM! Thank you for the update 👍 🚀

**Pull request merged**